spec/controllers/application_controller/ci_processing_spec.rb

```diff
@@ -120,9 +120,25 @@
                                         :controller             => "ems_cloud"
                                       )
       allow(controller).to receive(:drop_breadcrumb)
+      module ManageIQ::Providers::Example
+        class CloudManager < ManageIQ::Providers::CloudManager
+          def self.supports_discovery?
+            true
+          end
+
+          def self.ems_type
+            'example'
+          end
+
+          def self.description
+            'example Manager'
+          end
+        end
+      end
       controller.send(:discover)
       expect(response.status).to eq(200)
-      expect(controller.instance_variable_get(:@discover_type)).to include(["Azure", "azure"], ["Amazon", "amazon"])
+      expect(controller.instance_variable_get(:@discover_type)).to include(["example Manager", "example"])
+      ManageIQ::Providers.send(:remove_const, 'Example')
```

Comment:
@chrisarcand rspec :dollar: question: Do you happen to know how I can stub that `Example::CloudManager` ? Note that it needs to inherit from `Providers::CloudManager` so that `Provider::CloudManager.subclasses` returns `Example::CloudManager`